binderhub/registry.py

```diff
@@ -339,6 +339,39 @@ async def get_credentials(self, image, tag):
         return None
 
 
+class GoogleArtifactRegistry(DockerRegistry):
+    """
+    A registry for Google Artifact Registry.
+
+    At present, the only way to generate credentials without having to use the gcloud cli
+    is the metadata server. The metadata server uses the Compute Engine default service account.
+
+    For more information, see https://cloud.google.com/docs/authentication/rest#metadata-server.
+    """
+
+    @default("token_url")
+    def _default_token_url(self):
+        return "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token"
+
+    async def _get_token(self, client, token_url, service, scope):
+        auth_req = httpclient.HTTPRequest(
+            token_url, headers={"Metadata-Flavor": "Google"}
+        )
+        self.log.debug(
+            f"Getting registry token from {token_url} service={service} scope={scope}"
```

Comment:
Does the metadata server support the service/scope argument? If not we should omit it from the log.